src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,237 @@
+
+            self.__n_gallery = 0
+            self._gallery_index2imgid = {}
+            for img_id in self.__gallery_imgid2paths:
+                self._gallery_index2imgid[self._data_len + self.__n_gallery] = img_id
+                self.__n_gallery += 1
+
+            self._data_len += self.__n_gallery
+
+        self.__imgid2index = {}
+        for index, img_id in self.__index2imgid.items():
+            if img_id not in self.__imgid2index:
+                self.__imgid2index[img_id] = index
+
+        self.__scores, self.__is_query = self.__get_targets()
+
+    def __getitem__(self, index: int) -> dict:
+
+        if index < self.__n_queries + self.__n_relevant:
+            img_id = self.__index2imgid[index]
+            image_path = self._data_folder / self._imgid2paths[img_id]
+        else:
+            img_id = self._gallery_index2imgid[index]
+            image_path = self.__gallery_folder / self.__gallery_imgid2paths[img_id]
+
+        image = self._read_image(image_path)
+        sample = {'image': image}
+        sample = self._apply_transform(self.augment, sample)
+        sample = self._apply_transform(self.transform, sample)
+        sample['image'] = sample['image'].type(torch.__dict__[self._input_dtype])
+        sample['index'] = index
+        sample['is_query'] = self.__is_query[index]
+        sample['scores'] = self.__scores[index]
+
+        return sample
+
+    def __parse_match_csv(self):
+        self.__query_arr = self.__matches.loc[:, 'query'].tolist()
+        self.__index2imgid = dict(enumerate(self.__query_arr))
+        self.__n_queries = len(self.__index2imgid)
+
+        self.__relevant_arr, self.__relevance_scores = [], []
+        self.__n_relevant = 0
+
+        for index in range(len(self.__matches)):
+            self.__relevant_arr.append([])
+            self.__relevance_scores.append([])
+            rel_img_idxs = map(int, self.__matches.iloc[index]['relevant'].split())
+            rel_img_scores = map(float, self.__matches.iloc[index]['scores'].split())
+            for img_id, img_score in zip(rel_img_idxs, rel_img_scores):
+                # save unique image_id
+                if img_id not in self.__index2imgid.values():
+                    self.__index2imgid[self.__n_queries + self.__n_relevant] = img_id
+                    self.__n_relevant += 1
+                self.__relevant_arr[-1].append(img_id)
+                self.__relevance_scores[-1].append(img_score)
+
+    def __get_targets(self) -> Tuple[torch.FloatTensor, torch.BoolTensor]:
+        """Maps item scores to queues.
+
+        Returns:
+            Two target tensor: scores and is_query.
+            Scores is tensor with shape(len(self), n_queries).
+            Is_query is tensor with shape (len(self)).
+
+        Raises:
+            ValueError: If relevant objects list doesn't match with relevance scores list in size.
+        """
+
+        scores = torch.zeros((len(self), self.__n_queries), dtype=torch.float32)
```

Comment:
Please, explain me why it shape is `(len(self), self.__n_queries)` but not `(len(self), len(self))`.